### ISSUE TITLE
Why the loss function is focal loss?

### ISSUE THREAD

**Issue by Lin-plax** [ronghuaiyang/arcface-pytorch#105]
The arcface function is in the models/metrics.py, but the loss function is focal loss. In my mind, the arcface is a loss function. Can you tell me where is my worry?

**Comment by abcde-bit** [comment]
兄弟，这个问题解决了吗？我也想知道为什么是focal loss而不是arcface loss

**Comment by Lin-plax** [comment]
> 兄弟，这个问题解决了吗？我也想知道为什么是focal loss而不是arcface loss

arcface不是loss，是增加的一个惩罚项，在model/metric里面被定义，有可学习的参数，参与模型的参数更新。一般来说人脸学习是度量学习，这个arcface等于是将度量学习转换为分类学习了。

**Comment by abcde-bit** [comment]
明白了，谢谢大佬！！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ronghuaiyang/arcface-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月6日(星期一) 下午4:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ronghuaiyang/arcface-pytorch] Why the loss function is focal loss? (Issue #105)





  
兄弟，这个问题解决了吗？我也想知道为什么是focal loss而不是arcface loss
  
arcface不是loss，是增加的一个惩罚项，在model/metric里面被定义，有可学习的参数，参与模型的参数更新。一般来说人脸学习是度量学习，这个arcface等于是将度量学习转换为分类学习了。

**Comment by abcde-bit** [comment]
兄弟，可以留一个联系方式吗？我的QQ是2312428043




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ronghuaiyang/arcface-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月6日(星期一) 下午4:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ronghuaiyang/arcface-pytorch] Why the loss function is focal loss? (Issue #105)





  
兄弟，这个问题解决了吗？我也想知道为什么是focal loss而不是arcface loss
  
arcface不是loss，是增加的一个惩罚项，在model/metric里面被定义，有可学习的参数，参与模型的参数更新。一般来说人脸学习是度量学习，这个arcface等于是将度量学习转换为分类学习了。

**Comment by Lin-plax** [comment]
> 兄弟，可以留一个联系方式吗？我的QQ是2312428043
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "ronghuaiyang/arcface-pytorch" ***@***.***&gt;; 发送时间:&nbsp;2023年2月6日(星期一) 下午4:03 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [ronghuaiyang/arcface-pytorch] Why the loss function is focal loss? (Issue #105) 兄弟，这个问题解决了吗？我也想知道为什么是focal loss而不是arcface loss arcface不是loss，是增加的一个惩罚项，在model/metric里面被定义，有可学习的参数，参与模型的参数更新。一般来说人脸学习是度量学习，这个arcface等于是将度量学习转换为分类学习了。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

可以，一起交流
